Updating DXP space with illustations (#5242)
umbraco/UmbracoDocs: marketplace-and-integrations/README.md

Within this Umbraco DXP (Digital Experience Platform) area of the Umbraco documentation, you'll find topics related to extending Umbraco CMS via packages and integrations.

Umbraco is used in solutions in both a "platform" and "composable" DXP context. With the former, Umbraco has responsibility for both content management and delivery, with extensions added as NuGet packages. In a composable DXP, Umbraco is often a stand-alone service that takes part in an overall architecture. In these cases, Umbraco CMS is focused on the management of content exposed via APIs for delivery.

> Umbraco is also used in a hybrid DXP context. Here, some values are added to the Umbraco platform, like Umbraco Forms or Umbraco Commerce, where Umbraco acts as a component in a wider architecture. In all situations, backoffice integrations are an important part of providing an intuitive, unified editor experience.

Reviewer: ```suggestion
Umbraco is also used in a hybrid DXP context. Here, some services are added directly to the Umbraco platform, like Umbraco Forms or Umbraco Commerce, but Umbraco still acts as a component in a wider architecture.

In all situations, backoffice integrations with third-party services are an important part of providing an intuitive, unified editor experience.
```

Reviewer 2: How about this, to shorten the sentences a bit:

```suggestion
Umbraco is also used in a hybrid DXP context. Here, some services are added directly to the Umbraco platform while Umbraco still acts as a component in a wider architecture. This could be services like Umbraco Forms or Umbraco Commerce.
```

Reviewer: Not sure - on reflection we are overusing the word "services" here, so I've changed to "features" in my suggestion above.

If we break out the mention of Forms and Commerce into a separate sentence, it feels like we have to name them as packages again, which we are already referencing in the paragraph above this one.

So overall I think I prefer my modified suggestion above, as it's just there as a passing reference, making it clear what we are talking about when it comes to features in this context.

Our aim is to support the different ways of using Umbraco in a DXP context and provid resources for building integrations with Umbraco.

![Umbraco as a platform DXP, composable DXP og hybrid solution](.gitbook/assets/pDXPcDXP.png)

### Packages and integrations  

> Both packages and integrations are defined as Umbraco Packages that can be either [Nuget packages (Umbraco 10+) or packages zip files (Umbraco 8 or earlier)](https://docs.umbraco.com/umbraco-cms/extending/packages/types-of-packages).
> Integrations provide functionality via a third-party service's API, whereas packages extend the functionality of the CMS (management or delivery).

Reviewer: ```suggestion
Packages for Umbraco 10+ are installed via [Nuget packages](https://docs.umbraco.com/umbraco-cms/extending/packages/types-of-packages).

Integrations provide functionality via a third-party service's API, whereas packages extend the functionality of the CMS (management or delivery).
```

I'm not sure we really need to be calling out Umbraco 8 features any more, so think we can remove this part.

Reviewer 2: If we're not mentioning Umbraco 8, does it then make sense to mention Umbraco 10+?
We could just leave out the version numbers entirely.

![Ways to extend Umbraco](.gitbook/assets/PackagesAndIntegrations.png)

## Umbraco Commercial Products

Umbraco makes available three licensed add-ons that are available for purchase. They each have their dedicated documentation area:

> <table data-card-size="large" data-view="cards"><thead><tr><th></th><th></th><th data-hidden data-card-target data-type="content-ref"></th><th data-hidden data-card-cover data-type="files"></th></tr></thead><tbody><tr><td><strong>Umbraco Forms</strong></td><td>Extend your Umbraco CMS website with custom forms in all shapes and sizes.</td><td><a href="https://docs.umbraco.com/umbraco-forms">https://docs.umbraco.com/umbraco-forms</a></td><td><a href=".gitbook/assets/forms7.png">forms7.png</a></td></tr><tr><td><strong>Umbraco Deploy</strong></td><td>Synchronize content and code changes between your Umbraco website environments.</td><td><a href="https://docs.umbraco.com/umbraco-deploy">https://docs.umbraco.com/umbraco-deploy</a></td><td><a href=".gitbook/assets/paperplane7.png">paperplane7.png</a></td></tr><tr><td><strong>Umbraco Workflow</strong></td><td>Structure content reviews and approval flows for Content Editors on your website.</td><td><a href="https://docs.umbraco.com/umbraco-workflow">https://docs.umbraco.com/umbraco-workflow</a></td><td><a href=".gitbook/assets/Workflow7.png">Workflow7.png</a></td></tr><tr><td><strong>Umbraco Commerce</strong></td><td>Setup shop within a familiar structure and interface that is the Umbraco CMS with Umbraco Commerce.</td><td><a href="http://localhost:5000/o/vHdmkfI8smZW50A5yIZD/s/3rCehcwXc4nbpeGqcI2f/">12.commerce.latest (RC)</a></td><td><a href=".gitbook/assets/Untitled presentation.png">Untitled presentation.png</a></td></tr></tbody></table>

Reviewer: There's a link to `localhost:5000` in here, which was already there, that isn't correct.

Reviewer: @jonat123 - link issue was just this one here.

Within this Umbraco DXP (Digital Experience Platform) area of the Umbraco documentation, you'll find topics related to extending Umbraco CMS via packages and integrations.

Both packages and integrations are defined as Umbraco Packages that can be either [NuGet packages (Umbraco 10+) or packages zip files (Umbraco 8 or earlier)](https://docs.umbraco.com/umbraco-cms/extending/packages/types-of-packages)

Integrations are defined as Umbraco Packages that provide functionality via a third-party service's API, whereas packages are Umbraco Packages that extend the functionality of the CMS (management or delivery). For visualisation of ways to extend Umbraco see the illustation below.
![Packages and integrations](https://github.com/BoletteKern/UmbracoDocs/blob/main/marketplace-and-integrations/.gitbook/assets/Packages%20and%20intgerations.png)

> Umbraco is used in solutions in both a "platform" and "composable" DXP context. With the former, Umbraco has responsibility for both content management and delivery, with extensions added as NuGet packages. In a composable DXP, Umbraco is often a stand-alone service that take part of an overall architecture, focussed on the management of content exposed via APIs for delivery. Umbraco is also used in a hybrid DXP context where some values are added into the Umbraco platform, e.g Umbraco Forms or Umbraco Commerce, at the time as Umbraco is used as a component in a wider architecture. In all situations, backoffice integrations are an important part of providing an intuitive, unified editor experience.

Reviewer 2: We have some long sentences here, so I've tried to shorten them a bit.
Also separated the last bit, to break up the paragraph a little.

Let me know what you think. and if it doesn't work!

```suggestion
Umbraco is used in solutions in both a "platform" and "composable" DXP context. With the former, Umbraco has responsibility for both content management and delivery, with extensions added as NuGet packages. In a composable DXP, Umbraco is often a stand-alone service that takes part in an overall architecture. In these cases, Umbraco CMS is focused on the management of content exposed via APIs for delivery.

Umbraco is also used in a hybrid DXP context. Here, some values are added to the Umbraco platform, like Umbraco Forms or Commerce, where Umbraco acts as a component in a wider architecture. In all situations, backoffice integrations are an important part of providing an intuitive, unified editor experience.
```

![Umbraco as a platform DXP, composable DXP og hybrid solution](https://github.com/BoletteKern/UmbracoDocs/blob/main/marketplace-and-integrations/.gitbook/assets/pDXP%20cDXP.png)

> Our aim is to support all the different ways of using Umbraco in a DXP setup and make it simpler to build integrations with Umbraco.

Reviewer 2: A suggestion to avoid using the word "simpler":

```suggestion
Our aim is to support the different ways of using Umbraco in a DXP setup and provide resources for building integrations with Umbraco.
```

Within this Umbraco DXP (Digital Experience Platform) area of the Umbraco documentation, you'll find topics related to extending Umbraco CMS via packages and integrations.

> Both packages and integrations are defined as Umbraco Packages that can be either [NuGet packages (Umbraco 10+) or packages zip files (Umbraco 8 or earlier)](https://docs.umbraco.com/umbraco-cms/extending/packages/types-of-packages)

Reviewer: The content here is good, but I would say this distinction between "package" and "integration" is not to so important as the explanation of what we mean by a DXP in the context of Umbraco.  So I would suggest it goes below that, perhaps under a "Packages and integrations" sub-heading.

Obviously naming isn't great here, and it's a bit confusing that we talk about "packages" as anything you install into Umbraco via NuGet and "packages" as "those packages that aren't integrations".  Perhaps it could even just be the "integrations" that are called out for explicit definition, which, along with the diagram should illustrate what's important here.

Author: I agree! I removed the definition of packages and integrations but I don't know where to place it, because it's difficult to explain the diagram within the current structure of the page, which I like. Maybe we should consider simplifying the diagram before using it, or not using it at all. Let's talk about this Tuesday, and leave it out of the PR for now.
